### PR TITLE
[CI] Remove nested templates in the running steps.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Similar PR to https://github.com/dotnet/maui/pull/15365 

We want to remove the template explosion by moving to a matrix strategy.

### CODE REVIEW

File: eng/pipelines/common/maui-templates.yml [hunk 1]
```diff
@@ -125,64 +122,76 @@ jobs:
...
+    matrix:
+      ${{ each RunPlatform in parameters.RunPlatforms }}:
+        ${{ RunPlatform.testName }}:
+          TEST_NAME: ${{ RunPlatform.testName }}
+          VM_IMAGE: ${{ RunPlatform.vmImage }}
+          OS_VERSION: Ventura
+          OS_ARCH: x64
+
+  pool: 
+    vmImage: $(VM_IMAGE)
```
Comment (rmarinho) [inline eng/pipelines/common/maui-templates.yml:141]: this is missing - name: $(androidTestsVmPool) so we can say  `Azure Pipelines` or `Xamarin Shared`

**Review by rmarinho — CHANGES_REQUESTED** [review]

File: eng/pipelines/common/maui-templates.yml [hunk 2]
```diff
@@ -125,64 +122,78 @@ jobs:
...
+  condition: ${{ parameters.condition}}
+  strategy:
+    matrix:
+      ${{ each RunPlatform in parameters.RunPlatforms }}:
+        ${{ RunPlatform.testName }}:
+          TEST_NAME: ${{ RunPlatform.testName }}
+          VM_IMAGE: ${{ RunPlatform.vmImage }}
+          OS_VERSION: Ventura
+          OS_ARCH: x64
+          POOL_NAME: ${{ RunPlatform.name}}
```
Comment (mandel-macaque) [inline eng/pipelines/common/maui-templates.yml:139]:
```suggestion
          POOL_NAME: ${{ RunPlatform.name }}
```

File: eng/pipelines/common/maui-templates.yml [hunk 3]
```diff
@@ -125,64 +122,78 @@ jobs:
...
+          VM_IMAGE: ${{ RunPlatform.vmImage }}
+          OS_VERSION: Ventura
+          OS_ARCH: x64
+          POOL_NAME: ${{ RunPlatform.name }}
+
+  pool: 
+    name: $(POOL_NAME)
+    vmImage: $(VM_IMAGE)
+    demands:
+      - macOS.Name -equals $(OS_VERSION)
```
Comment (rmarinho) [inline eng/pipelines/common/maui-templates.yml:145]: These demands are conditional should be a variable to this case only for the android pool they are used.

**Review by rmarinho — CHANGES_REQUESTED** [review]

File: eng/pipelines/common/maui-templates.yml (hunk 3 above)
Comment (mandel-macaque) [reply eng/pipelines/common/maui-templates.yml:145]: Issue is, it cannot be a condition without being a template.. I'll try to think a way, but it is an issue.

**Comment by mandel-macaque** [comment]
This is actually an impossible change to make due to how the demands in pools work.

**Pull request closed without merging**